Source/items.cpp

```diff
@@ -420,42 +420,47 @@ void InitItemGFX()
 	memset(UniqueItemFlags, 0, sizeof(UniqueItemFlags));
 }
 
-bool ItemPlace(int xp, int yp)
+bool ItemPlace(Point position)
 {
-	if (dMonster[xp][yp] != 0)
+	if (dMonster[position.x][position.y] != 0)
 		return false;
-	if (dPlayer[xp][yp] != 0)
+	if (dPlayer[position.x][position.y] != 0)
 		return false;
-	if (dItem[xp][yp] != 0)
+	if (dItem[position.x][position.y] != 0)
 		return false;
-	if (dObject[xp][yp] != 0)
+	if (dObject[position.x][position.y] != 0)
 		return false;
-	if ((dFlags[xp][yp] & BFLAG_POPULATED) != 0)
+	if ((dFlags[position.x][position.y] & BFLAG_POPULATED) != 0)
 		return false;
-	if (nSolidTable[dPiece[xp][yp]])
+	if (nSolidTable[dPiece[position.x][position.y]])
 		return false;
 
 	return true;
 }
 
+Point GenerateValidRandomItemPosition()

```

Comment:
I would also say that Generate is the wrong verb for getting an enum value, it would more be fore generating some sort of sequence. Also the negation of Valid is Invalid which I don't see as a possible option for an alternative function, so maybe Available instead. Lastly Random appears to just be an implementation detail.

Maybe `GetAvalibleItemPosition()`,  `GetAvaliblePosition()`,  `GetAvalibleGroundPosition()`?

Reply:
> Generate is the wrong verb for getting an enum value

"enum value"? Did you mean something else?

> Also the negation of Valid is Invalid which I don't see as a possible option for an alternative function, so maybe Available instead. 

Fair. Will change to `Available`.

> Lastly Random appears to just be an implementation detail.

Also fair, good point.

> `GetAvaliblePosition()`, `GetAvalibleGroundPosition()`

Those would be too general and thus misleading. The implementation is specific to items, there are other similar ones for monsters and objects (which we can combine later). I'll go with your first suggestion.

Reply:
Actually this seams some what linked to level generation and so the random calls are crucial, so maybe `GetRandomAvalibleItemPosition`?

Reply:
Renamed again.